lib/rules/indent.js

```diff
@@ -125,43 +123,48 @@ const KNOWN_NODES = new Set([
 
 
 /**
- * A mutable balanced binary search tree that stores (key, value) pairs. The keys are numeric, and must be unique.
- * This is intended to be a generic wrapper around a balanced binary search tree library, so that the underlying implementation
+ * A mutable map that stores (key, value) pairs. The keys are numeric indices, and must be unique.
+ * This is intended to be a generic wrapper around a map with non-negative integer keys, so that the underlying implementation
  * can easily be swapped out.
  */
-class BinarySearchTree {
+class IndexMap {
 
     /**
-     * Creates an empty tree
+     * Creates an empty map
+     * @param {number} maxKey The maximum key
      */
-    constructor() {
-        this._orderedMap = new OrderedMap();
-        this._orderedMapEnd = this._orderedMap.end();
+    constructor(maxKey) {
+
+        // Initializing the array with the maximum expected size avoids dynamic reallocations that could degrade performance.
+        this._values = Array(maxKey + 1);
     }
 
     /**
-     * Inserts an entry into the tree.
+     * Inserts an entry into the map.
      * @param {number} key The entry's key
      * @param {any} value The entry's value
      * @returns {void}
      */
     insert(key, value) {
-        this._orderedMap.setElement(key, value);
+        this._values[key] = value;
     }
 
     /**
-     * Finds the entry with the largest key less than or equal to the provided key
+     * Finds the value of the entry with the largest key less than or equal to the provided key
      * @param {number} key The provided key
-     * @returns {{key: number, value: *}|null} The found entry, or null if no such entry exists.
+     * @returns {number|undefined} The value of the found entry, or undefined if no such entry exists.
```

Comment:
```suggestion
     * @returns {*|undefined} The value of the found entry, or undefined if no such entry exists.
```

This should be `*` because it returns value, not key?

Reply:
Thanks! Yes, that's the value.